x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/textembedding/TextEmbeddingInternalService.java

```diff
@@ -333,6 +338,38 @@ public void putModel(Model model, ActionListener<Boolean> listener) {
         }
     }
 
+    @Override
+    public void isModelDownloaded(Model model, ActionListener<Boolean> listener) {
+        ActionListener<GetTrainedModelsAction.Response> getModelsResponseListener = listener.delegateFailure((delegate, response) -> {
+            if (response.getResources().count() < 1) {
+                delegate.onResponse(Boolean.FALSE);
+            } else {
+                delegate.onResponse(Boolean.TRUE);
+            }
+        });
+
+        // TODO move modelId to TextEmbeddingModel to avoid checking for these model subtypes
```

Comment:
Yeah I think that'd be a cleaner solution. Any reason you're holding off doing that in this PR?

Reply:
I was able to make this change, and I added a test for the problem I was running into. I think it would be considerably harder to test for the double download case that Dave mentioned, so I didn't add any tests for that.